src/random/sampling.c

```diff
@@ -95,17 +95,73 @@ ret_t sample_uniform_r_bits_with_fixed_prf_context(
   return SUCCESS;
 }
 
+#if defined (AVX2)
+
 _INLINE_ int is_new(IN const idx_t *wlist, IN const size_t ctr)
 {
-  for(size_t i = 0; i < ctr; i++) {
-    if(wlist[i] == wlist[ctr]) {
-      return 0;
+    REG_T complist, current, comp;
+    uint32_t check;
+    current = SET1_I32(wlist[ctr]);
+    for(size_t i = 0; i < (ctr/DWORDS_IN_YMM); i++)
+    {
+        complist = LOAD(&wlist[i * DWORDS_IN_YMM]); 
+        comp = CMPEQ_I32(current, complist);
+        check = MOVEMASK(comp);
+
+        if(check != 0)
```

Comment:
Please run `tests/run_tests.sh 0` before committing code (as noted in Readme). This will run all the sanitizers. For example, an error is reported here because the if statement is not inside braces.